src/actions/public/getTransaction.ts

```diff
@@ -34,7 +34,7 @@ export type GetTransactionParameters =
       blockHash?: never
       blockNumber?: never
       /** The block tag. */
-      blockTag: BlockTag
+      blockTag: TBlockTag
```

Comment:
One thing I like doing with generics is adding the type along with the generic value at the infer site.

```suggestion
      blockTag: TBlockTag | BlockTag
```

This generally makes the editor UI a bit better, by displaying all possible values instead of just the inferred value. For example, autocomplete:

| Before | After |
|--------|--------|
| <img width="523" alt="image" src="https://github.com/wagmi-dev/viem/assets/6759464/31cb3971-a37e-49f7-a134-465d872bc184"> | <img width="516" alt="image" src="https://github.com/wagmi-dev/viem/assets/6759464/39bd4934-b0a6-4460-ac83-b69094ad9b8f"> |

And hover:

| Before | After |
|--------|--------|
| <img width="463" alt="image" src="https://github.com/wagmi-dev/viem/assets/6759464/e09826e3-b075-4dce-be7e-3a4beb3691be"> | <img width="460" alt="image" src="https://github.com/wagmi-dev/viem/assets/6759464/fa2d193d-0a1c-4255-a2d1-d457c595eb03"> | 

Can see another example of this in https://github.com/wagmi-dev/viem/pull/857